internal/controllers/app_controller.go

```diff
@@ -224,6 +224,76 @@ func (r appReconcileResult) isConflictError() bool {
 	}
 }
 
+type condition struct {
+	Type   string
+	Reason string
+}
+
+// workload contains the needed information for watchDeployEvents logic
+// deployments and statefulsets are both supported so it became necessary
+// to abstract their common properties into a separate type
+type workload struct {
+	Name               string
+	Replicas           int
+	UpdatedReplicas    int
+	ReadyReplicas      int
+	Generation         int
+	ObservedGeneration int
+	Conditions         []condition
+}
+
+type workloadClient struct {
+	workloadType      ketchv1.AppType
+	workloadName      string
+	workloadNamespace string
+	k8sClient         kubernetes.Interface
+}
+
+// Get populates workload values based on the workloadType and returns the populated struct
+func (cli workloadClient) Get(ctx context.Context) (*workload, error) {
+	switch cli.workloadType {
+	case "Deployment":
```

Comment:
would you mind using ketchv1.DeploymentAppType and ketchv1.StatefulSetAppType there?

Reply:
Done